rust/operator-binary/src/controller.rs

```diff
@@ -839,48 +835,46 @@ fn build_server_rolegroup_statefulset(
 }
 
 fn add_s3_volume_and_volume_mounts(
-    s3_conn: Option<&S3ConnectionSpec>,
+    s3_conn: &S3ConnectionSpec,
     cb_druid: &mut ContainerBuilder,
     pb: &mut PodBuilder,
 ) -> Result<()> {
-    if let Some(s3_conn) = s3_conn {
-        if let Some(credentials) = &s3_conn.credentials {
-            const VOLUME_NAME: &str = "s3-credentials";
-            pb.add_volume(credentials.to_volume(VOLUME_NAME).context(
-                CredentialsToVolumeSnafu {
+    if let Some(credentials) = &s3_conn.credentials {
```

Comment:
Oh I wasn't aware edc is using S3... In that case this needs a bigger re-work, which I'm happy to do (after I finished the other S3 tools)